packages/protocol/contracts/common/MentoFeeHandlerSeller.sol

```diff
@@ -0,0 +1,110 @@
+  using FixidityLib for FixidityLib.Fraction;
+
+  /**
+   * @notice Sets initialized == true on implementation contracts.
+   * @param test Set to true to skip implementation initialisation.
+   */
+  constructor(bool test) public Initializable(test) {}
+
+  function initialize(address _registryAddress) external initializer {
+    _transferOwnership(msg.sender);
+    setRegistry(_registryAddress);
+  }
+
+  /**
+   * @notice Returns the storage, major, minor, and patch version of the contract.
+   * @return Storage version of the contract.
+   * @return Major version of the contract.
+   * @return Minor version of the contract.
+   * @return Patch version of the contract.
+   */
+  function getVersionNumber() external pure returns (uint256, uint256, uint256, uint256) {
+    return (1, 1, 0, 0);
+  }
+
+  function sell(
+    address sellTokenAddress,
+    address buyToken,
+    uint256 amount,
+    uint256 maxSlippage // as fraction,
+  ) external {
+    require(
+      buyToken == registry.getAddressForOrDie(GOLD_TOKEN_REGISTRY_ID),
+      "Buy token can only be gold token"
+    );
+
+    StableToken stableToken = StableToken(sellTokenAddress);
+
+    address exchangeAddress = registry.getAddressForOrDie(stableToken.getExchangeRegistryId());
+
+    IExchange exchange = IExchange(exchangeAddress);
+
+    uint256 minAmount = 0;
+    if (maxSlippage != 0) {
+      // max slippage is set
+      // use sorted oracles as reference
+
+      // TODO check amount of reports or that the bucket hasn't been updated in 5 minutes
+      // safetyCheck() or modifier
+      ISortedOracles sortedOracles = getSortedOracles();
+      (uint256 rateNumerator, uint256 rateDenominator) = sortedOracles.medianRate(sellTokenAddress);
+      minAmount = calculateMinAmount(rateNumerator, rateDenominator, amount, maxSlippage);
+    }
+
+    // TODO an upgrade would be to compare using routers as well
+    stableToken.approve(exchangeAddress, amount);
+    exchange.sell(amount, minAmount, false);
+
+    IERC20 goldToken = getGoldToken();
+    // IERC20 stableAsERC20 = IERC20(sellTokenAddress);
+    goldToken.transfer(msg.sender, goldToken.balanceOf(address(this)));
+
+  }
+
+  // TODO move this to abstract class
+  function calculateMinAmount(
+    uint256 midPriceNumerator,
+    uint256 midPriceDenominator,
+    uint256 amount,
+    uint256 maxSlippage // as fraction
+  )
+    public
+    pure
+    returns (
+      // FixidityLib.Fraction memory maxSlippage
+      uint256
+    )
+  {
+    FixidityLib.Fraction memory maxSlippageFraction = FixidityLib.wrap(maxSlippage);
+
+    FixidityLib.Fraction memory price = FixidityLib.newFixedFraction(
+      midPriceNumerator,
+      midPriceDenominator
+    );
+    FixidityLib.Fraction memory amountFraction = FixidityLib.newFixed(amount);
+    FixidityLib.Fraction memory totalAmount = price.multiply(amountFraction);
+
+    return
+      totalAmount
+        .subtract((price.multiply(maxSlippageFraction)).multiply(amountFraction))
```

Comment:
Nit: we can remove the inner brackets

.subtract(price.multiply(maxSlippageFraction).multiply(amountFraction))